#! /bin/sh /usr/share/dpatch/dpatch-run
## 13_arm_gcc4.dpatch by Brian Nelson <pyro@debian.org> by way of Jeremy
## Laine <jeremy.laine@m4x.org>.
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes FTBFS on arm

@DPATCH@
diff -urNad qt4-x11-4.0.1~/src/corelib/tools/qchar.h qt4-x11-4.0.1/src/corelib/tools/qchar.h
--- qt4-x11-4.0.1~/src/corelib/tools/qchar.h	2005-08-16 11:41:08.000000000 +0300
+++ qt4-x11-4.0.1/src/corelib/tools/qchar.h	2005-12-13 15:32:45.000000000 +0200
@@ -219,7 +219,7 @@
     inline const char toLatin1() const;
     inline const ushort unicode() const { return ucs; }
 #ifdef Q_NO_PACKED_REFERENCE
-    inline ushort &unicode() { return *(&ucs); }
+    inline ushort &unicode() { return *((ushort*)&ucs); }
 #else
     inline ushort &unicode() { return ucs; }
 #endif
